lms/static/scripts/frontend_apps/components/EmailDigestPreferences.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button, Checkbox } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import { useCallback } from 'preact/hooks';
+
+import type { SelectedDays, WeekDay } from '../config';
+
+const dayNames: [WeekDay, string][] = [
+  ['sun', 'Sunday'],
+  ['mon', 'Monday'],
+  ['tue', 'Tuesday'],
+  ['wed', 'Wednesday'],
+  ['thu', 'Thursday'],
+  ['fri', 'Friday'],
+  ['sat', 'Saturday'],
+];
+
+export type EmailDigestPreferencesProps = {
+  /** Currently selected days */
+  selectedDays: SelectedDays;
+  /** Callback to fully or partially update currently selected days, without saving */
+  onSelectedDaysChange: (newSelectedDays: Partial<SelectedDays>) => void;
+};
+
+export default function EmailDigestPreferences({
+  selectedDays,
+  onSelectedDaysChange,
+}: EmailDigestPreferencesProps) {
+  const setAllTo = useCallback(
+    (enabled: boolean) =>
+      onSelectedDaysChange({
+        sun: enabled,
+        mon: enabled,
+        tue: enabled,
+        wed: enabled,
+        thu: enabled,
+        fri: enabled,
+        sat: enabled,
+      }),
+    [onSelectedDaysChange],
+  );
+  const selectAll = useCallback(() => setAllTo(true), [setAllTo]);
+  const selectNone = useCallback(() => setAllTo(false), [setAllTo]);
+
+  return (
+    <div className="flex flex-col gap-y-3">
+      <h2 className="text-lg/4 text-brand">Student activity digest</h2>
+      <p>Receive email notifications when your students annotate.</p>
+      <p>Select the days you{"'"}d like your emails:</p>
+
+      <div className="flex justify-between pl-4">
+        <div className="flex flex-col gap-1">
+          {dayNames.map(([day, name]) => (
+            <span
+              key={day}
+              className={classnames(
+                // The checked icon sets fill from the text color
+                'text-grey-6',
+              )}
+            >
```

Comment:
The amount of markup needed to customize the checkbox icon fill here, and then undo it for the label, is unfortunate. This is also duplicated for every checkbox in this UI. If the `Checkbox` class had a way to customize the classes applied to just the icon, this would make things easier.